phalcon/config/factory.zep

```diff
@@ -49,7 +49,15 @@ class Factory extends BaseFactory
 
 	protected static function loadClass(string $namespace, var config)
 	{
-		var adapter, className, mode, callbacks, filePath;
+		var adapter, className, mode, callbacks, filePath, extension;
+
+		if typeof config == "string" {
+			let extension = pathinfo(config, PATHINFO_EXTENSION);
```

Comment:
If you want only the file extension, use this:
```zep
let extension = substr(strrchr(config, "."), 1);
```
This is many times faster than using `pathinfo()`.

Reply:
```php
$startTime=microtime(true);
for($i = 0; $i < 10000; $i++) {
    $test = substr(strrchr($config, "."), 1);
}
var_dump(microtime(true)-$startTime);
$startTime=microtime(true);
for($i = 0; $i < 10000; $i++) {
    pathInfo($config, PATHINFO_EXTENSION);
}
var_dump(microtime(true)-$startTime);
```

```
/vagrant/www/test.php:81:float 0.036841869354248
/vagrant/www/test.php:86:float 0.02786111831665
```

It's actually not.

Reply:
Yea in zephir it's faster.